garnett/posfilewriter.py

```diff
@@ -95,10 +95,8 @@ def _write(msg, end='\n'):
 
             # shape defs
             try:
-                required = set(frame.types).intersection(
-                    set(frame.shapedef.keys()))
-                not_defined = set(frame.types).difference(
-                    set(frame.shapedef.keys()))
+                required = [t for t in frame.types if t in frame.shapedef]
```

Comment:
This rewritten code may be less robust in a few edge cases where there are undefined types, or more type names than type_shapes, etc. It handles those cases less well because the specification with `type_shapes` is a little more rigorously defined. I think we can accept it as-is if tests pass, and handle those edge cases if users need them. I would suspect those would only be seen in very old pre-HOOMD POS files.

Reply:
What I meant to ask was why were you using the `shapedef` property instead of the `type_shapes`, since `shapedef` will be removed in the future. But I can see your point now.